### ISSUE TITLE
Expand section on code generation - generate from Dart vs generate from interface description

### ISSUE THREAD

**Issue by natebosch** [matanlurey/dart_serialize_proposal#2]
For serialization where the messages are strictly values (think a typed version of the `Map` approach) and don't have behavior associated with them there is a choice of describing the structure with Dart code as shown, or describing the structure completely outside of Dart and generating the classes entirely.

The latter is the approach I'm using [here](https://github.com/natebosch/dart_language_server/blob/master/tool/message_builder.dart) where I start with a yaml description of the interfaces.

**Comment by matanlurey** [comment]
Yep, could you add an example of your YAML file and a short blurb explaining the approach?
